rasa/core/validator.py

```diff
@@ -0,0 +1,178 @@
+import logging
+import asyncio
+from typing import List, Any, Text
+from rasa.core.domain import Domain
+from rasa.nlu.training_data import load_data, TrainingData
+from rasa.core.training.dsl import StoryFileReader, StoryStep
+from rasa.core.training.dsl import UserUttered
+from rasa.core.training.dsl import ActionExecuted
+
+logger = logging.getLogger(__name__)
+
+
+class Validator(object):
+    """A class used to verify usage of intents and utterances."""
+
+    def __init__(self, domain: Domain, intents: TrainingData, stories: List[StoryStep]):
+        """Initializes the Validator object. """
+
+        self.domain = domain
+        self.intents = intents
+        self.valid_intents = []
+        self.valid_utterances = []
+        self.stories = stories
+
+    @classmethod
+    def from_files(
+        cls, domain_file: Text, nlu_data: Text, story_data: Text
+    ) -> "Validator":
+        """Create an instance from the domain, nlu and story files."""
+
+        domain = Domain.load(domain_file)
+        loop = asyncio.new_event_loop()
+        stories = loop.run_until_complete(
+            StoryFileReader.read_from_folder(story_data, domain)
+        )
+        intents = load_data(nlu_data)
+        return cls(domain, intents, stories)
+
+    def _search(self, vector: List[Any], searched_value: Any) -> bool:
+        """Searches for a element in a vector."""
```

Comment:
I don't think we need this function, python can already handle list searching -- check my comment in verify_intents